aci/resource_aci_cloudaccount.go

```diff
@@ -0,0 +1,368 @@
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+	}
+
+	if relationTocloudRsCredentials, ok := d.GetOk("relation_cloud_rs_credentials"); ok {
+		relationParam := relationTocloudRsCredentials.(string)
+		err = aciClient.CreateRelationcloudRsCredentials(cloudAccount.DistinguishedName, cloudAccountAttr.Annotation, relationParam)
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+	}
+
+	d.SetId(cloudAccount.DistinguishedName)
+	log.Printf("[DEBUG] %s: Creation finished successfully", d.Id())
+	return resourceAciAccountRead(ctx, d, m)
+}
+
+func resourceAciAccountUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] Account: Beginning Update")
+	aciClient := m.(*client.Client)
+	account_id := d.Get("account_id").(string)
+	vendor := d.Get("vendor").(string)
+	TenantDn := d.Get("tenant_dn").(string)
+
+	cloudAccountAttr := models.AccountAttributes{}
+
+	nameAlias := ""
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		nameAlias = NameAlias.(string)
+	}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		cloudAccountAttr.Annotation = Annotation.(string)
+	} else {
+		cloudAccountAttr.Annotation = "{}"
+	}
+
+	if AccessType, ok := d.GetOk("access_type"); ok {
+		cloudAccountAttr.AccessType = AccessType.(string)
+	}
+
+	if Account_id, ok := d.GetOk("account_id"); ok {
+		cloudAccountAttr.Account_id = Account_id.(string)
+	}
+
+	if Name, ok := d.GetOk("name"); ok {
+		cloudAccountAttr.Name = Name.(string)
+	}
+
+	if Vendor, ok := d.GetOk("vendor"); ok {
+		cloudAccountAttr.Vendor = Vendor.(string)
+	}
+	cloudAccount := models.NewAccount(fmt.Sprintf("act-[%s]-vendor-%s", account_id, vendor), TenantDn, nameAlias, cloudAccountAttr)
+
+	cloudAccount.Status = "modified"
+
+	err := aciClient.Save(cloudAccount)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	checkDns := make([]string, 0, 1)
+
+	if d.HasChange("relation_cloud_rs_account_to_access_policy") || d.HasChange("annotation") {
+		_, newRelParam := d.GetChange("relation_cloud_rs_account_to_access_policy")
+		checkDns = append(checkDns, newRelParam.(string))
+
+	}
+
+	if d.HasChange("relation_cloud_rs_credentials") || d.HasChange("annotation") {
+		_, newRelParam := d.GetChange("relation_cloud_rs_credentials")
+		checkDns = append(checkDns, newRelParam.(string))
+
+	}
+
+	d.Partial(true)
+	err = checkTDn(aciClient, checkDns)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.Partial(false)
+
+	if d.HasChange("relation_cloud_rs_account_to_access_policy") || d.HasChange("annotation") {
+		_, newRelParam := d.GetChange("relation_cloud_rs_account_to_access_policy")
+		err = aciClient.DeleteRelationcloudRsAccountToAccessPolicy(cloudAccount.DistinguishedName)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		err = aciClient.CreateRelationcloudRsAccountToAccessPolicy(cloudAccount.DistinguishedName, cloudAccountAttr.Annotation, newRelParam.(string))
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
```

Comment:
shouldn't there be a condition on the recreation? What happens if the relationship is not defined anymore?